lib/vnet/setup_daemon_darwin.go

```diff
@@ -36,3 +36,7 @@ func execAdminProcess(ctx context.Context, config daemon.Config) error {
 
 	return trace.Wrap(execAdminSubcommand(ctx, config))
 }
+
+func DaemonSubcommand(ctx context.Context) error {
+	return trace.Wrap(daemon.Start(ctx, AdminSetup))
```

Comment:
I feel like it'd be better if the daemon explicitly called `AdminSetup`. But that's not possible because of how packages are organized – `lib/vnet/daemon` is imported by `lib/vnet` already for `daemon.Config`.

I'm not sure how worthwhile refactoring this is. I'd need to create a separate `lib/vnet/setup` package with `AdminSetup`, so that `lib/vnet` and `lib/vnet/daemon` can call this function directly. Then I'd also likely need another package which holds the equivalent of `daemon.Config`, so that `setup`, `daemon` and `vnet` can operate on a common struct.

Reply:
My 2c is that it's worth thinking about your packages import relationships and refactor accordingly. I suggest doing it in a follow up, there's enough here already (and a follow up with just moves is relatively simple to review).